x/bank/module.go

```diff
@@ -107,10 +107,10 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
 }
 
 // NewAppModule creates a new AppModule object.
-func NewAppModule(cdc codec.Marshaler, keeper keeper.Keeper, accountKeeper types.AccountKeeper) AppModule {
+func NewAppModule(cdc codec.Codec, keeper keeper.Keeper, accountKeeper types.AccountKeeper) AppModule {
 	return AppModule{
 		AppModuleBasic:  AppModuleBasic{},
-		cosmosAppModule: sdkbank.NewAppModule(cdc, keeper, accountKeeper),
+		cosmosAppModule: sdkbank.NewAppModule(cdc, keeper.BaseKeeper, accountKeeper),
```

Comment:
why do we unwrap this?

Reply:
There is a migration logic in the base bank module that involves type assertion to the `BaseKeeper`. Not unwrapping the keeper leads to a panic due to mismatched types.